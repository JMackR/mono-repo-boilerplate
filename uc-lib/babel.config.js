module.exports = {
  presets: [["@babel/preset-env", { targets: { node: "current" } }], "module:metro-react-native-babel-preset"],
  plugins: ["@babel/plugin-proposal-class-properties", "inline-json-import"],
  sourceType: "unambiguous",
}
