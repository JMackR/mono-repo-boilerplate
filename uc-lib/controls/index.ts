export * from "./image"
export * from "./button"
export * from "./text"
export * from "./activity-indicator"
export * from "./separator"
export * from "./list"
export * from "./section-list"
export * from "./badge"
export * from "./toggle"
export * from "./login-button"
export * from "./layout"
export * from "./background"
export * from "./grid"
export * from "./touchable"
export * from "./view-block"
export * from "./text-entry"
export * from "./scroll-view"
export * from "./circle"
export * from "./slider"
export * from "./clickable"
export * from "./map"
export * from "./visibility-check"
export * from "./strike-through-text"
// export * from "./stripe-text-entry"
export * from "./input"
