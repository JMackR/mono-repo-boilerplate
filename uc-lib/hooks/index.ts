export * from "./use-infinite-scroll"
export * from "./selectable"
export * from "./screen-provider"
export * from "./loading"
