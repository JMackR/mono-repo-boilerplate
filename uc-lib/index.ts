export * from "./animations"
export * from "./assets"
export * from "./controls"
export * from "./themes"
export * from "./widgets"
export * from "./utilities"
export * from "./hooks"
export * from "./validators"
