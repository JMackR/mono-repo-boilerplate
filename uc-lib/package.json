{
  "name": "redibs-ucl",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "test": "jest --projects test/react test/react-native test/non-react"
  },
  "dependencies": {
    "@types/ramda": "^0.27.14",
    "classnames": "2.2.6",
    "google-map-react": "1.1.7",
    "hoist-non-react-statics": "3.3.2",
    "imask": "6.0.5",
    "react-compound-slider": "2.5.0",
    "react-native-autolink": "3.0.0",
    "react-native-fs": "^2.16.6",
    "zxcvbn": "4.4.2"
  },
  "devDependencies": {},
  "prettier": {
    "printWidth": 125,
    "semi": false,
    "singleQuote": false,
    "trailingComma": "all"
  },
  "jest-junit": {
    "outputDirectory": "../build",
    "ancestorSeparator": " â€º ",
    "classNameTemplate": "{classname}",
    "suiteName": "redibs-redibs-ucl jest tests",
    "suiteNameTemplate": "{filename}",
    "titleTemplate": "{title}",
    "uniqueOutputName": "true"
  }
}
