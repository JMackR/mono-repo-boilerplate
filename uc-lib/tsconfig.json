{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "lib",
    "rootDir": "../",
    "jsx": "react-native",
    "composite": true,
    "declarationMap": true,
    "isolatedModules": true,
    "sourceMap": true,
    "moduleResolution": "Node",
    "esModuleInterop": true,
    "typeRoots": ["../node_modules/@types", "./types"]
  },
  "include": ["./"],
  "exclude": ["node_modules", "babel.config.js", "metro.config.js", "jest.config.js", "types"]
}
