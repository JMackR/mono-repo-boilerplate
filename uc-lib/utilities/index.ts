export * from "./dimensions-utility"
export * from "./media-query"
