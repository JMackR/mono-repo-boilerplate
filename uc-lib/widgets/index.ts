export * from "./account"
export * from "./attribute-list"
export * from "./breadcrumb"
export * from "./dialog/"
export * from "./empty-state"
export * from "./expanding-text-section"
export * from "./flexible-row"
export * from "./full-screen-spinner"
export * from "./inbox-row"
export * from "./indicator-dots"
export * from "./item-preview"
export * from "./listing-photo-list"
export * from "./pill-list"
export * from "./segmented-control"
export * from "./selectable-row"
export * from "./tooltip"
export * from "./user-profile"
export * from "./search"
