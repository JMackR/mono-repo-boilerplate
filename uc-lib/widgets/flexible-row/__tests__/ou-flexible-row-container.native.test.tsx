import { FlexibleRow, FlexibleRowContainer } from "uc-lib"
import React from "react"
import { render } from "react-native-testing-library"

describe("FlexibleRow Snapshot Tests", () => {
  test("Render", () => {
    const tree = render(
      <FlexibleRowContainer>
        <FlexibleRow mainContent="Text" clickAction={() => {}} />
      </FlexibleRowContainer>,
    )
    expect(tree).toMatchSnapshot()
  })
  test("Render with extended margins", () => {
    const tree = render(
      <FlexibleRowContainer extendMargin={true}>
        <FlexibleRow mainContent="Text" clickAction={() => {}} />
      </FlexibleRowContainer>,
    )
    expect(tree).toMatchSnapshot()
  })
})
