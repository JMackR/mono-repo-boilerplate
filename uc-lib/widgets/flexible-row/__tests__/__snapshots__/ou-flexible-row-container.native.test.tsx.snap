// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexibleRow Snapshot Tests Render 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": undefined,
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexGrow": 0,
        "flexShrink": 1,
        "height": undefined,
        "justifyContent": "flex-start",
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#E2E2E2",
          "borderBottomWidth": 0.5,
          "borderLeftColor": "#E2E2E2",
          "borderLeftWidth": 0.5,
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    accessibilityLabel="redibs-ucl.flexible-row"
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
        "opacity": 1,
      }
    }
    testID="redibs-ucl.flexible-row"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "height": undefined,
          "minHeight": 42,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <Text
          accessibilityLabel="redibs-ucl.flex-row.main-content"
          style={
            Array [
              Object {
                "fontFamily": "Mulish-Regular",
                "fontSize": 16,
                "fontWeight": "normal",
                "lineHeight": 20,
              },
              Object {
                "color": "#121212",
                "textAlign": undefined,
                "textDecorationLine": undefined,
              },
            ]
          }
          testID="redibs-ucl.flex-row.main-content"
        >
          Text
        </Text>
      </View>
      <View
        accessibilityLabel="redibs-ucl.flex-row.right-arrow"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
        testID="redibs-ucl.flex-row.right-arrow"
      >
        <actions-chevron-right-line.svg
          alwaysDark="#121212"
          alwaysLight="#FFFFFF"
          basalt="#8A8A8A"
          crystal="#FFFFFF"
          crystalHover="#F7F7F7"
          crystalPressed="#EEEEEE"
          disabled="#FAFAFA"
          glacialBlue="#3D98D2"
          granite="#ABABAB"
          larchYellow="#F4BE19"
          larchYellowHover="#F5D624"
          larchYellowPressed="#F3A70C"
          limestone="#E2E2E2"
          limestoneHover="#EEEEEE"
          limestonePressed="#D0D0D0"
          obsidian="#121212"
          overlay="rgba(18,18,18,0.3)"
          paintbrushRed="#E05666"
          green="#00A87E"
          greenHighlight="#DEF3EC"
          greenHover="#43B893"
          greenPressed="#00986A"
          quartz="#FAFAFA"
          tintColor="#ABABAB"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": undefined,
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexGrow": 0,
        "flexShrink": 1,
        "height": undefined,
        "justifyContent": "flex-start",
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#E2E2E2",
          "borderBottomWidth": 0.5,
          "borderLeftColor": "#E2E2E2",
          "borderLeftWidth": 0.5,
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`FlexibleRow Snapshot Tests Render with extended margins 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": undefined,
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexGrow": 0,
        "flexShrink": 1,
        "height": undefined,
        "justifyContent": "flex-start",
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#E2E2E2",
          "borderBottomWidth": 0.5,
          "borderLeftColor": "#E2E2E2",
          "borderLeftWidth": 0.5,
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    accessibilityLabel="redibs-ucl.flexible-row"
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
        "opacity": 1,
      }
    }
    testID="redibs-ucl.flexible-row"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "height": undefined,
          "minHeight": 42,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <Text
          accessibilityLabel="redibs-ucl.flex-row.main-content"
          style={
            Array [
              Object {
                "fontFamily": "Mulish-Regular",
                "fontSize": 16,
                "fontWeight": "normal",
                "lineHeight": 20,
              },
              Object {
                "color": "#121212",
                "textAlign": undefined,
                "textDecorationLine": undefined,
              },
            ]
          }
          testID="redibs-ucl.flex-row.main-content"
        >
          Text
        </Text>
      </View>
      <View
        accessibilityLabel="redibs-ucl.flex-row.right-arrow"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
        testID="redibs-ucl.flex-row.right-arrow"
      >
        <actions-chevron-right-line.svg
          alwaysDark="#121212"
          alwaysLight="#FFFFFF"
          basalt="#8A8A8A"
          crystal="#FFFFFF"
          crystalHover="#F7F7F7"
          crystalPressed="#EEEEEE"
          disabled="#FAFAFA"
          glacialBlue="#3D98D2"
          granite="#ABABAB"
          larchYellow="#F4BE19"
          larchYellowHover="#F5D624"
          larchYellowPressed="#F3A70C"
          limestone="#E2E2E2"
          limestoneHover="#EEEEEE"
          limestonePressed="#D0D0D0"
          obsidian="#121212"
          overlay="rgba(18,18,18,0.3)"
          paintbrushRed="#E05666"
          green="#00A87E"
          greenHighlight="#DEF3EC"
          greenHover="#43B893"
          greenPressed="#00986A"
          quartz="#FAFAFA"
          tintColor="#ABABAB"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": undefined,
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexGrow": 0,
        "flexShrink": 1,
        "height": undefined,
        "justifyContent": "flex-start",
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#E2E2E2",
          "borderBottomWidth": 0.5,
          "borderLeftColor": "#E2E2E2",
          "borderLeftWidth": 0.5,
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
